python/ray/serve/config.py

```diff
@@ -698,3 +702,24 @@ class Config:
         validate_assignment = True
         extra = "forbid"
         arbitrary_types_allowed = True
+
+
+@PublicAPI(stability="beta")
+class gRPCOptions(BaseModel):
+    port: int = -1
```

Comment:
why not set the default port instead of -1?

Reply:
Bc by default we don't want gRPC proxy to start. Defaulting to a valid port is misleading

Reply:
ic, is it any downside we make a feature flag to decide whether to start gRPC proxy? so that `gRPCOptions` can hold the default values, user can get it working by enable the flag directly without touching the option.

Reply:
Yea, there's a choice to be made. We can offer some default gRPC port, but also note that it does nothing without user defining their `grpc_servicer_functions` (since no services and methods will be added if nothing is defined). I think the best is to allow this option with the intention of using gRPC else it's just wasting resources having a gRPC server running without a way to use it.

Feature flag is fine, but since the intention of using the options already serve as the gate, I think it only makes things more complicated for the users (in that if they want to use gRPC, they have to specify the gRPC options + setting the feature flag vs in the current implementation just specify the gRPC option).